doc/barclamp.md

```diff
@@ -0,0 +1,1177 @@
+data bag from the previous section. Similar to the previous section we will
+start out with a copy of the Barbican barclamp's schema
+(`template-barbican.schema`), which we'll copy to `template-mybarclamp.json`.
+
+Again we will first globally search and replace `barbican` by `` and
+then go through it section by section, starting with the top-level fields. The
+only thing that needs adjusting at the top level is the pattern constraint for
+the `id` field, which our global search and replace will already have taken
+care of, leaving us with a top-level section that should look something like this:
+
+```
+{
+  "required": true,
+  "type": "map",
+  "mapping": {
+    "id": { "type": "str", "required": true, "pattern": "/^-|^template-mybarclamp$/" },
+    "description": { "required": true, "type": "str" },
+    "attributes": {
+      "required": true,
+      "type": "map",
+      "mapping": {
+        "": {
+```
+
+The corresponding section from the Barbican data bag schema begins
+[here](https://github.com/crowbar/crowbar-openstack/blob/8bebf8a379ebea8ef462ad49746dda6d36a3c46d/chef/data_bags/crowbar/template-barbican.schema#L1) and ends
+[here](https://github.com/crowbar/crowbar-openstack/blob/8bebf8a379ebea8ef462ad49746dda6d36a3c46d/chef/data_bags/crowbar/template-barbican.schema#L11).
+
+Next, we will need to adjust the constraints for `attributes[""]`. The Crowbar
+web UI is a Ruby on Rails application with a MVC (Model-View-Controller)
+architecture so you will need to create the MVC architecture's three
+components for your barclamp plus some glue and metadata. This section will
+take you through all of these steps.
+
+##### Barclamp Metadata
+
+Location: `mybarclamp.yml`
+
+Much like the chef cookbook, the Crowbar application needs some metadata about
+your barclamp. This is a YAML formatted file residing in the top level
+directory. It's filename consists of your barclamp's name plus the extension
+`.yml`. Below you will find a commented example:
+
+```
+barclamp:
+  # Your barclamp's name as it appears in file names.
+  name: 'mybarclamp'
+  # Your barclamp's name as it will be displayed in the Crowbar web UI.
+  display: 'Mybarclamp'
+  # Here we usually have the service name and a short description of the
+  # Service the barclamp deploys for OpenStack barclamps.
+  description: 'OpenStack Myservice: Provides Foo and Bar'
+  # Just leave this as it is.
+  version: 0
+  # This is a repeat of the requirements list from the chef cookbook.
+  requires:
+    - '@crowbar'
+    - 'database'
+    - 'rabbitmq'
+    - 'keystone'
+  # This denotes the Barclamp collection this barclamp belongs to.
+  member:
+    - 'openstack'
+
+crowbar:
+  # The barclamp's priority in the crowbar UI (lower number means higher priority)
+  run_order: 98
+  # The barclamp's priority in the ordering of chef cookbooks (lower number
+  # means higher priority)
+  chef_order: 98
+
+  # FIXME: The following appear to be unused. They are neither loaded in
```

Comment:
I'm 85% sure that this comment is misleading and that at least one if not two or all three of these are in use.